stripe/src/main/java/com/stripe/android/model/PaymentIntent.java

```diff
@@ -300,11 +370,65 @@ public JSONObject toJson() {
         map.put(FIELD_DESCRIPTION, mDescription);
         map.put(FIELD_LIVEMODE, mLiveMode);
         map.put(FIELD_NEXT_SOURCE_ACTION, mNextSourceAction);
+        map.put(FIELD_NEXT_ACTION, mNextAction);
         map.put(FIELD_RECEIPT_EMAIL, mReceiptEmail);
         map.put(FIELD_STATUS, mStatus);
         map.put(FIELD_SOURCE, mSource);
         StripeNetworkUtils.removeNullAndEmptyParams(map);
         return map;
     }
 
+    public enum NextActionType {
+        RedirectToUrl("redirect_to_url"),
+        AuthorizeWithUrl("authorize_with_url");
+
+        @NonNull public final String code;
+
+        NextActionType(@NonNull String code) {
+            this.code = code;
+        }
+
+        @Nullable
+        public static NextActionType fromCode(@Nullable String code) {
+            if (code == null) {
+                return null;
+            }
+
+            for (NextActionType nextActionType : values()) {
+                if (nextActionType.code.equals(code)) {
+                    return nextActionType;
+                }
+            }
+
+            return null;
+        }
+    }
+
+    public enum Status {
+        RequiresSource("requires_source"),
```

Comment:
in an ideal world we would also cut out these old statuses, but i recognize that may not be possible